src/Security/UserChecker.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace AppBundle\Security;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Exception\AccountNotValidatedException;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\User\UserCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class UserChecker implements UserCheckerInterface
+{
+    private $router;
+
+    public function __construct(Router $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkPreAuth(UserInterface $user)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkPostAuth(UserInterface $user)
+    {
+        /** @var Adherent $user */
+        if (!$user instanceof Adherent && $user->isEnabled()) {
```

Comment:
Let's throw a RuntimeException if it's not an Adherent instance no ?
It would be more accurate.

Reply:
done

Reply:
Ok but now the second test `$user->isEnabled()` should be on its own no and throw the `DisabledException` no ?